[Evals] Add hard Chinese translations

nicognaW opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, **failure to follow the guidelines below will result in the PR being closed automatically**. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access be granted. 🚨

**PLEASE READ THIS**:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject it since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, please note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑

### Eval name

chinese_hard_translations

### Eval description

Evaluate the model's ability of understanding hard-to-understand Chinese sentences.

### What makes this a useful eval?

It improves the accuracy of Chinese-English translation. There's a set of Chinese sentences (mostly from Weibo, the Chinese social media) that have been pointed out by Chinese NLP researchers that are extremely hard to understand. So I collected some and tested them with GPT-4 with plugins, filtering out some of them that GPT-4 could understand accurately. Here is the case-by-case explanation of the first 5 cases:

### 一位友好的哥谭市民

This one is actually from a real Chinese news show, in this case, "的哥" is a slang meaning text drivers, but "哥谭" is the Chinese translation of Gotham, the city name from Batman. Machine translation systems misunderstand it as the city name rather than the slang. This sentence both fails on GPT-4 & Google Translate:

![GPT-4](https://github.com/openai/evals/assets/66731869/e504254f-e1c2-4dc7-992c-10d0fa9c9744)
![Google](https://github.com/openai/evals/assets/66731869/a2816995-6ab1-4f8c-bb82-85cf4a04f494)

### 中不不建交是受印度的影响，中不建交不受印度的影响。

In this sentence, "不" is short for the country name Bhutan. GPT-4 fails while Google Translate succeeds:

![GPT-4](https://github.com/openai/evals/assets/66731869/5d66dfe9-aa36-4a47-b567-f9ac3421f585)
![Google](https://github.com/openai/evals/assets/66731869/9b71877b-e06f-4a4b-aeac-fb2ad0b45253)

### 骑车出门差点摔跤，还好我一把把把把住了

Translation systems tend to miss the meaning (handlebar) of the word "把".I tested this twice, and one succeeds and one fails:

![GPT-4](https://github.com/openai/evals/assets/66731869/44c66feb-8644-48d8-9103-c764774869dd)
![GPT-4](https://github.com/openai/evals/assets/66731869/b782abe6-7d86-457b-9605-1c1649d2b12b)

Google fails BTW:
![Google](https://github.com/openai/evals/assets/66731869/08614acc-fb2a-4fb5-bffb-44cbeeba3bf9)

### 我背有点驼，妈妈说“你的背得背背背背佳

"背背佳" is a name of a Chinese product, and I have confirmed that the GPT model has the correct knowledge about this product:

![GPT-3.5](https://github.com/openai/evals/assets/66731869/5f0b10da-6e91-48f2-a9f6-691fa433228b)

And it fails to translate this sentence:

![GPT-4](https://github.com/openai/evals/assets/66731869/2977dd09-7e97-4746-9cf9-9504866c919b)

### 祝你全家幸福！

This is a tricky one. In Chinese, when people say your family, they include you as part of your family, which is equal to "you and your family" when we talk in English. Google Translate does it well, while GPT-4 doesn't capture this subtle difference:

![GPT-4](https://github.com/openai/evals/assets/66731869/24455e17-da27-4362-9e95-8da6dd555fc5)
![Google](https://github.com/openai/evals/assets/66731869/84b26baf-eaab-4002-ad72-de65e48154d5)

### Others

I collected most of the cases from weibo, especially from Dr.Qun Liu, great thanks to @liuquncn .

**All of the cases fail on GPT-4, and if you are confused about any of them, please leave a comment and I will explain then.**

### Evaluation result

Here is the final report evaluating `gpt-3.5-turbo` on my local:

```json
{"final_report": {"accuracy": 0.25, "sacrebleu_score": 28.24146086212546}}
```

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high-quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should

- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your YAML is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (<https://platform.openai.com/docs/usage-policies>).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the commits on the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgment

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and the high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access be granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "一位友好的哥谭市民"}], "ideal": ["Citizen Tan, a friendly taxi driver"]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "中不不建交是受印度的影响。"}], "ideal": ["The establishment of diplomatic relations between China and Bhutan is influenced by India."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "骑车差点摔跤，还好我一把把把把住了"}], "ideal": ["I almost fell off my bike, but luckily I managed to hold onto the handlebar."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "我背有点驼，妈妈说“你的背得背背背背佳"}], "ideal": ["I have a bit of a hunchback, and mom said, \"You need to use BeiBeiJia for your back.\""]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "祝你全家幸福！"}], "ideal": ["I wish you and your family happiness!"]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "郭麒麟刷牙选中华为的就是干净，速度快。"}], "ideal": ["Guo Qilin chose Zhonghua toothpaste for brushing his teeth because it's clean and fast."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "广州人和牲畜配种中心"}], "ideal": ["Guangzhou Renhe Livestock Breeding Center"]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "数学王子秦老师"}], "ideal": ["Mathematics teacher Wang Ziqin"]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "男子跳河轻生河水没过脖子。"}], "ideal": ["A man attempted suicide by jumping into a river, the water rose above his neck."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "一位漂亮的女孩说：今日本美女面色蜡黄，不适合上班。"}], "ideal": ["A beautiful girl said: Today, this beauty looks pale and is not fit for work."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "我们家热得快炸了。"}], "ideal": ["Our Fast Heat at home broke down."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "你去班上数数数数数不好的有多少"}], "ideal": ["Go to the class and count how many can't even count properly."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "只有小孩才说喝茶，成人用品"}], "ideal": ["Only children say \"drink tea\", adults say \"taste tea\"."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "汉黄香九岁温席奉亲。"}], "ideal": ["Huang Xiang of the Han Dynasty warmed her parents' quilt with her body."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "华强答应郝哥买瓜。"}], "ideal": ["Huaqiang promised Brother Hao to buy melons."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "这车是卖书的的。"}], "ideal": ["This vehicle belongs to the book seller."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "你什么时候都可以来，但就是我方便的时候不能来。"}], "ideal": ["You can come anytime, but you can't come when I go to the toilet."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "没有秃顶的人没有秃顶的人厉害。"}], "ideal": ["People who are not bald are not as good as people who are bald."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "请不同意不下课的同学不要不举手。"}], "ideal": ["Please don't raise your hands if you disagree with not ending the class."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "海水朝朝朝朝朝朝朝落。"}], "ideal": ["The seawater rises with the tide, rising every morning, rising and falling every day."]}
{"input": [{"role": "system", "content": "Given a text representing, provide the English translation of the text. You **MUST NOT** provide any explanation in the output other than the translation itself. You **MUST** paraphrase rather than translate word for word, with **ALL** of the original meanings preserved."}, {"role": "user", "content": "浮云长长长长长长长消。"}], "ideal": ["The floating clouds spread, often spreading everywhere, frequently dissipating and fading away."]}
  ```
</details>

nicognaW commented:
Hi, @usama-openai. Could you please help me review this PR? I saw you are actively reviewing PRs. Thanks!

usama-openai commented:
> Hi, @usama-openai. Could you please help me review this PR? I saw you are actively reviewing PRs. Thanks!

This PR was draft PR earlier. That's why I didn't review this PR. I'll review it as soon as possible.

nicognaW commented:
Thanks! Look forward to your feedback.

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request a few changes:

1. For some samples, proper translation is not provided in the ideal answer. For example, consider the following sample:

    `一位漂亮的女孩说：今日本美女面色蜡黄，不适合上班`

    The ideal answer for this is, `Today, this beauty looks pale and is not fit for work.`. It seems like the translation for `一位漂亮的女孩说：` is missing from the ideal answer. Kindly add proper translation for all the samples.

2. Keep the sentences shorter than 15 words because there can be multiple ways of translating a sentence, and with larger sentences, there is a great chance that the model translation will differ from the ideal answer while having the same meaning.

We would love to review the PR again after the suggested changes.

nicognaW commented:
I shorted the sentence, and removed 2 samples that GPT-4 could understand accurately.

Added 4 samples, 1 is from traditional Chinese comedy, 3 are inspired by a modern Chinese linguist.

I'm still trying to make the samples useful.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.

usama-openai commented:
You should see GPT-4 API access enabled in your account in the next few days.
